minion-gateway/main/src/main/java/org/opennms/miniongateway/grpc/server/tasktresults/TaskResultsKafkaForwarder.java

```diff
@@ -37,7 +58,42 @@ public SinkModule<Message, Message> getModule() {
     public void handleMessage(Message messageLog) {
         logger.debug("Received results; sending to Kafka: kafka-topic={}, message={}", kafkaTopic, messageLog);
 
+        // Retrieve the Tenant ID from the TenantID GRPC Interceptor
+        String tenantId = tenantIDGrpcInterceptor.readCurrentContextTenantId();
+
+        if (tenantId == null) {
+            usingDefaultTenantIdLog.warn("!!! USING DEFAULT TENANT ID !!! (TASKSET RESULTS)");
+            tenantId = "opennms-prime";
+        }
+
         byte[] rawContent = messageLog.toByteArray();
-        this.kafkaTemplate.send(kafkaTopic, rawContent);
+
+        ProducerRecord<String, byte[]> producerRecord = formatProducerRecord(rawContent, tenantId);
+
+        this.kafkaTemplate.send(producerRecord);
+    }
+
+//========================================
+// INTERNALS
+//----------------------------------------
+
+    /**
+     * Format the record to send to Kafka, with the needed content and the headers.
+     *
+     * @param rawContent content to include as the message payload.
+     * @param tenantId Tenant ID to include in the message headers.
+     * @return ProducerRecord to send to Kafka.
+     */
+    private ProducerRecord<String, byte[]> formatProducerRecord(byte[] rawContent, String tenantId) {
+        List<Header> headers = new LinkedList<>();
+        headers.add(new RecordHeader(TENANT_ID_HEADER_NAME, tenantId.getBytes(StandardCharsets.UTF_8)));
+
+        return new ProducerRecord<String, byte[]>(
```

Comment:
there is another way to set headers without using nulls 
```suggestion
       var producerRecord = new ProducerRecord<String, byte[]>(kafkaTopic, rawContent);
        producerRecord.headers().add(new RecordHeader(TENANT_ID, tenantId.getBytes(StandardCharsets.UTF_8)));
```